pkg/controller/cheinstallation/cheinstallation_controller_test.go

```diff
@@ -440,16 +462,31 @@ func TestCreateNamespaceForChe(t *testing.T) {
 		cl, r := configureClient(t, cheInstallation, newCheNamespace(cheOperatorNS, v1.NamespaceActive))
 
 		// when
-		created, err := r.ensureCheNamespace(testLogger(), cheInstallation)
+		requeue, err := r.ensureCheNamespace(testLogger(), cheInstallation)
 
 		// then
 		require.NoError(t, err)
-		assert.False(t, created)
+		assert.False(t, requeue)
 		AssertThatNamespace(t, Namespace, cl).
 			Exists().
 			HasLabels(toolchain.Labels())
 	})
 
+	t.Run("should not fail as ns is in termination state", func(t *testing.T) {
+		// given
+		cheInstallation := NewInstallation()
+		cheOperatorNS := cheInstallation.Spec.CheOperatorSpec.Namespace
+		cl, r := configureClient(t, cheInstallation, newCheNamespace(cheOperatorNS, v1.NamespaceTerminating))
```

Comment:
wouldn't be more intuitive if the phase was set to an empty string? The termination state is not a realistic and correct state that we could expect. I know that it's only for the testing purposes, but it just looks weird

Reply:
sure, I could do that. But the problem we got elsewhere was with a namespace being deleted and the test failing to wait until the NS was recreated. Or I guess we have both cases, after all.

Reply:
actually, the NS can only be in 2 phases: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/namespaces.md#phases so I guess we should only use the `Terminating` phase in this state, even if it looks weird.

Reply:
Ok :+1: